src/mbgl/renderer/painter.hpp

```diff
@@ -178,7 +177,7 @@ class Painter : private util::noncopyable {
     }();
 
 private:
-    TransformState state;
+    Transform& transform;
```

Comment:
Can we revert the changes to painter.hpp/cpp and painter_*.cpp? We're trying to reduce shared state through references, and using `Transform&` reverses that.